common/persistence/operationModeValidator.go

```diff
@@ -135,9 +135,9 @@ func ValidateConflictResolveWorkflowModeState(
 	switch mode {
 	case ConflictResolveWorkflowModeUpdateCurrent:
 		// it is ok that currentWorkflowMutation is null, for 2 DC
+		// Note: current workflow mutation can be in zombie state, for the update
 		if resetWorkflowState == WorkflowStateZombie ||
-			(newWorkflowState != nil && *newWorkflowState == WorkflowStateZombie) ||
-			(currentWorkflowState != nil && *currentWorkflowState == WorkflowStateZombie) {
```

Comment:
Why current workflow will not be zombie?

Reply:
this validation is checking whether the status to be written down to DB make sense.

current workflow, if present, can be terminated or zombiefied so target workflow and / or new workflow can be persisted.
since the policy is `ConflictResolveWorkflowModeUpdateCurrent` so the target and / or new workflow must be non zombie

Reply:
the removal of check `(currentWorkflowState != nil && *currentWorkflowState == WorkflowStateZombie)` basically current workflow ending up being a zombie workflow. 
this should be allowed since current workflow may not be generated by this cluster